### DISCUSSION TITLE
RQL query for comparing two properties of thing Json

### DISCUSSION THREAD

**Discussion by BHUVANESHWARI967** [eclipse-ditto/ditto#1958] [Q&A]
Hello Everyone,

I am trying to compare two JSON properties of a thing to check if they are equal or not and consume events based on that condition.

I have tried:
`"_/_/things/twin/events?namespaces=vehicle&filter=ne(features/lightSensor/properties/state,features/lightSensor/desiredProperties/state)`"

but got an invalid RQL expression. How can I consume this event? Can you please suggest a way for this? Thank you.

**Comment by thjaeckle** [comment]
This is currently not possible to do.

**Comment by BHUVANESHWARI967** [comment]
Oh okay. So what can be an alternative for this?

**Reply by thjaeckle** [reply]
The RQL only can take concrete values on the right side of the comma.
For doing it completely dynamic based on 2 property values, there is no alternative at the moment.

Feel free to create an issue for that and also a PR if this is something you need from Ditto. 👍
